### PR TITLE
Add a tool-info module for Kissat SAT solver

### PR DESCRIPTION
**Pull request by nianzelee**
This PR provides a tool-info module for Kissat SAT solver.

### CODE REVIEW

File: benchexec/tools/kissat.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def determine_result(self, run):
+        """
+        @return: status of Kissat after executing a run
+        """
+
+        status = None
+
+        for line in run.output:
+            if "s SATISFIABLE" in line:
+                status = "SAT"
```
Comment (PhilippWendler) [inline benchexec/tools/kissat.py:39]:
```suggestion
                status = result.RESULT_TRUE_PROP
```
BenchExec does not support `SAT` and `UNSAT` anymore. You can use it, but such results will always be marked as "error" (e.g., pink in tables) and BenchExec cannot compare them against the expected verdict.

File: benchexec/tools/kissat.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        return tool_locator.find_executable("kissat", subdir="build")
+
+    def name(self):
+        return "Kissat"
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def cmdline(self, executable, options, task, rlimits):
+        return [executable] + options + list(task.input_files_or_identifier)
```
Comment (PhilippWendler) [inline benchexec/tools/kissat.py:28]: I guess that Kissat always needs exactly one input file? In this case, just use `[task.single_input_file]` as last part of the command line.

File: benchexec/tools/kissat.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        @return: status of Kissat after executing a run
+        """
+
+        status = None
+
+        for line in run.output:
+            if "s SATISFIABLE" in line:
+                status = "SAT"
+            elif "s UNSATISFIABLE" in line:
+                status = "UNSAT"
```
Comment (PhilippWendler) [inline benchexec/tools/kissat.py:41]:
```suggestion
                status = result.RESULT_FALSE_PROP
```

File: benchexec/tools/kissat.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        if (not status or status == result.RESULT_UNKNOWN) and run.was_timeout:
+            status = "TIMEOUT"
+
```
Comment (PhilippWendler) [inline benchexec/tools/kissat.py:45]:
```suggestion
```
These lines could be removed, BenchExec will apply the same logic internally anyway.

**Review by PhilippWendler — COMMENTED** [review]
Thanks, looks good, just a few things.

File: benchexec/tools/kissat.py (hunk 2 above)
Comment (nianzelee) [reply benchexec/tools/kissat.py:28]: Done.

File: benchexec/tools/kissat.py (hunk 1 above)
Comment (nianzelee) [reply benchexec/tools/kissat.py:39]: Done.

File: benchexec/tools/kissat.py (hunk 3 above)
Comment (nianzelee) [reply benchexec/tools/kissat.py:41]: Done.

File: benchexec/tools/kissat.py (hunk 4 above)
Comment (nianzelee) [reply benchexec/tools/kissat.py:45]: Done.

**Review by PhilippWendler — APPROVED** [review]
Thanks!

**Pull request merged**